src/foundation.cs

```diff
@@ -8780,6 +8784,27 @@ partial interface NSItemProvider : NSCopying {
 #endif
 	}
 
+	interface INSItemProviderReading {}
+	
+	[Watch (4,0), TV (11,0), Mac (10,13), iOS (11,0)]
+	[Protocol]
+	interface NSItemProviderReading
+	{
+		[Static, Abstract]
+		[Export ("readableTypeIdentifiersForItemProvider", ArgumentSemantic.Copy)]
+		string[] ReadableTypeIdentifiersForItemProvider { get; }
+
+		//
+		// This is a constructor that various classes must implement
+		// NSAttributedString, UIColor and UIImage need to have this constructor
+		// for user-defined implementations of this interface, we are going to
+		// need to something special
+		//
+		//[Abstract]
+		//[Export ("initWithItemProviderData:typeIdentifier:error:")]
+		//INSItemProviderReading CreateFrom (NSData providerData, string typeIdentifier, [NullAllowed] NSError outError);
```

Comment:
What this patch does is to inline the constructor in all three places that conform to this procotol.

I tried doing what you suggested (a) removing the above line, and (b) removing the inlined constructors that this patch contains everywhere and (c) making those classes adopt the NSItemProviderReading interface, but that does not inline the constructor anywhere, nor does the interface gain anything out of having this method declared here. 

The result is thus a step backwards, the patch to that is here:

https://gist.github.com/migueldeicaza/e22f44007e7ac093a0076276187dcb16

So I am not sure what is that you are suggesting I do with this.